src/me/EtienneDx/RealEstate/Config.java

```diff
@@ -212,4 +220,105 @@ public void loadConfig()
+
+            if (messageData.notes != null) {
+                messageData.notes = config.getString("Messages." + messageID.name() + ".Notes", messageData.notes);
+                config.set("Messages." + messageID.name() + ".Notes", messageData.notes);
+            }
+        }
+
+        //save any changes
+        try {
+            config.options().header("Use a YAML editor like NotepadPlusPlus to edit this file.  \nAfter editing, back up your changes before reloading the server in case you made a syntax error.  \nUse dollar signs ($) for formatting codes, which are documented here: http://minecraft.gamepedia.com/Formatting_codes");
+            config.save(messagesFilePath);
+        } catch (IOException exception) {
+            RealEstate.instance.log.info("Unable to write to the configuration file at \"" + messagesFilePath + "\"");
+        }
+
+        defaults.clear();
+        RealEstate.instance.log.info("Customizable messages loaded.");
+        System.gc();
+    }
+
+    private void addDefault(HashMap<String, CustomizableMessage> defaults,
+                            Messages id, String text, String notes) {
+        CustomizableMessage message = new CustomizableMessage(id, text, notes);
+        defaults.put(id.name(), message);
+    }
+
+    synchronized public String getMessage(Messages messageID, String... args) {
+        String message = messages[messageID.ordinal()];
+
+        for (int i = 0; i < args.length; i++) {
+            String param = args[i];
+            message = message.replace("{" + i + "}", param);
+        }
+
+        return message;
+    }
+    //sends a color-coded message to a player
+    public static void sendMessage(Player player, ChatColor color, Messages messageID, String... args) {
+        sendMessage(player, color, messageID, 0, args);
+    }
+
+    //sends a color-coded message to a player
+    public static void sendMessage(Player player, ChatColor color, Messages messageID, long delayInTicks, String... args) {
+        String message = RealEstate.instance.config.getMessage(messageID, args);
+        sendMessage(player, color, message, delayInTicks);
+    }
+
+    //sends a color-coded message to a player
+    public static void sendMessage(Player player, ChatColor color, String message) {
+        if (message == null || message.length() == 0) return;
+
+        if (player == null) {
+            RealEstate.instance.log.info(color + message);
+        } else {
+            player.sendMessage(RealEstate.instance.config.chatPrefix + color + message);
+        }
+    }
+
+    public static void sendMessage(Player player, ChatColor color, String message, long delayInTicks) {
+        SendPlayerMessageTask task = new SendPlayerMessageTask(player, color, message);
+
+        if (delayInTicks > 0) {
+            RealEstate.instance.getServer().getScheduler().runTaskLater(RealEstate.instance, task, delayInTicks);
+        } else {
+            task.run();
+        }
+    }

```

Comment:
I don't really see the point of this method, I mean it could be nice to send delayed messages, but I don't really see the applications?

Reply:
Sometimes an instruction might have a delay before it triggers (like Ess /home command with delay or GP /trapped command or whatever) and if we're sending a confirmation message right after the player run such commands it looks weird at least. In such case, (having async messaging approach) we would like to add some delay before send the message.

Not sure, that it's still applicable in this plugin, though I used this feature in other solutions.

Reply:
good enough for me, we'll have to check wether we use it or not but we can keep it for now ^^